src/osd/PG.cc

```diff
@@ -6549,6 +6547,7 @@ boost::statechart::result
 PG::RecoveryState::RepWaitBackfillReserved::react(const RemoteReservationRejected &evt)
 {
   PG *pg = context< RecoveryMachine >().pg;
+  pg->osd->remote_reserver.cancel_reservation(pg->info.pgid);
```

Comment:
This part looks good to me, but the commit message left me very confused since the statements are only valid when we start out in RepWaitBackfillReserved. ;)